rfc/0002_scope_manager.md

```diff
@@ -0,0 +1,61 @@
+
+# Technical background
+
+For any thread, at most one **Span** may be "active". Of course there may be many other **Spans** involved with the thread which are (a) started, (b) not finished, and yet (c) not "active": perhaps they are waiting for I/O, blocked on a child **Span**, or otherwise off of the critical path.
+ 
+It's inconvenient to pass an active **Span** from function to function manually, so OpenTracing should require that every **Tracer** contains a **Scope Manager** that grants access to the active **Span** through a container, called **Scope**.
+
+# Specification Changes
+
+New `ScopeManager` and `Scope` interfaces are added to the specification, and the `Tracer` interface is extended to support creation of `Span`s that are automatically set as the **active** one for the current context.
+
+## ScopeManager
+
+The `ScopeManager` interface allows setting the active `Span` in a call-context storage section, and has the following members:
+
+* **activate**, capability to set the specified `Span` as the active one for the current call-context, returning a `Scope` containing it. A parameter (still to decide whether it will be optional or required) will mark whether the returned `Scope` should, upon deactivation, **finish** the contained `Span`.
+* **active**, the `Scope` containing the current active `Span` if any, or else null/nothing.
+
+## Scope
+
+The `Scope` interface acts as a container of the active `Span` for the current-call context, and has the following members:
+
+* **span**, the contained active `Span` for this call-context. It will never be null/nothing.
+* **close**, marking the end of the active period for the current `Span`, and optionally **finishing** it. Calling it more tan once leads to undefined behavior.
+
+If the language supports some kind of auto finishing statement (such as `try` for Java, or `with` for Python), `Scope` should adhere to such convention. Additionally, `Scope` is not guaranteed to be thread-safe.
+
+## Tracer changes
+
+The `Tracer` interface will be extended with:
+
+* **scope manager**, the `ScopeManager` tracking the active `Span` for this instance.
+* **start active span**, a new behavior for starting a new `Span`, which will be automatically marked as active for the current call-context. It will return a `Scope`.
+* Both **start span** and **start active span** will implicitly use any active `Span` as the parent for newly created `Span`s (with a `ChildOf` relationship), unless the parent is explicitly specified, or (the new) **ignore parent span** parameter is specified (in which case the resulting `Span` will have no parent at all).
```

Comment:
"ignore **active** span" instead of "ignore **parent** span". (The method name in Java/C# is `ignoreActiveSpan`).